lib/miq_ldap.rb

```diff
@@ -46,8 +46,11 @@ def initialize(options = {})
     options[:host] ||= ::Settings.authentication.ldaphost
     options[:port] ||= ::Settings.authentication.ldapport
     options[:host] = resolve_host(options[:host], options[:port])
-    options[:encryption] = mode == "ldaps" ? {:method => :simple_tls} : {}
-    options.store_path(:encryption, :tls_options, :verify_mode, OpenSSL::SSL::VERIFY_NONE) if options[:host].ipaddress?
+
+    if mode == "ldaps"
+      options[:encryption] = {:method => :simple_tls}
```

Comment:
Does `options[:encryption]` need to be a hash even when not using ldaps?  I ask because it used to be an empty hash.

Reply:
No, options[:encryption] does not need to be a hash even when not using ldaps.

In fact this PR is fixing exactly that. options[:encryption] can not be an empty hash if it is not needed because net-ldap chokes on and empty hash for options[:encryption] 

Making it an empty hash was the bug I introduced with https://github.com/ManageIQ/manageiq/pull/16850
that this PR is fixing.

Initially options[:encryption] was only set if mode == "ldaps" as follows:

`options[:encryption] = {:method => :simple_tls} if mode == "ldaps"`

Reply:
@bdunne Thanks again!